Export Analysis Reports/Sdl.Community.ExportAnalysisReports/Service/ReportService.cs

```diff
@@ -0,0 +1,355 @@
+						{
+							if (string.IsNullOrEmpty(project.ReportPath))
+							{
+								project.ReportPath = reportOutputPath;
+							}
+
+							WriteReportFile(project, optionalInformation, languageReport, isChecked);
+						}
+					}
+				}
+				return true;
+			}
+			catch (Exception exception)
+			{
+				Log.Logger.Error($"GenerateReport method: {exception.Message}\n {exception.StackTrace}");
+				throw;
+			}
+		}
+
+		/// <summary>
+		/// Get report output path from Json file
+		/// </summary>
+		/// <param name="jsonPath"></param>
+		/// <returns></returns>
+		public string GetJsonReportPath(string jsonPath)
+		{
+			if (File.Exists(jsonPath))
+			{
+				using (var r = new StreamReader(jsonPath))
+				{
+					var json = r.ReadToEnd();
+					var item = JsonConvert.DeserializeObject<JsonSettings>(json);
+					if (!string.IsNullOrEmpty(item.ExportPath))
+					{
+						return item.ExportPath;
+					}
+				}
+			}
+			return string.Empty;
+		}
+
+		/// <summary>
+		/// Check if the exported path is the same
+		/// </summary>
+		/// <param name="reportOutputPath"></param>
+		/// <returns></returns>
+		public bool IsSameReportPath(string reportOutputPath)
+		{
+			var jsonReportPath = GetJsonReportPath(JsonPath);
+			if (!string.IsNullOrEmpty(jsonReportPath) && !string.IsNullOrEmpty(reportOutputPath) && jsonReportPath.Equals(reportOutputPath))
+			{
+				return true;
+			}
+
+			return false;
+		}
+
+		/// <summary>
+		/// Check if the report file exists
+		/// </summary>
+		/// <param name="reportFolderPath"></param>
+		/// <returns></returns>
+		public bool ReportFileExist(string reportFolderPath)
+		{
+			try
+			{
+				if (!string.IsNullOrEmpty(reportFolderPath))
+				{
+					var fileName = Path.GetFileName(Path.GetDirectoryName(reportFolderPath));
+					if (Directory.Exists(reportFolderPath))
+					{
+						var files = Directory.GetFiles(reportFolderPath);
+						if (files.Any(file => file.Contains("Analyze Files")))
+						{
+							return true;
+						}
+
+						_messageBoxService.ShowInformationMessage(string.Format(PluginResources.ExecuteAnalyzeBatchTask_Message, fileName), PluginResources.InformativeLabel);
+						return false;
+					}
+
+					if (!string.IsNullOrEmpty(fileName) && fileName.Contains("ProjectFiles"))
+					{
+						fileName = Path.GetFileNameWithoutExtension(fileName);
+						return !string.IsNullOrEmpty(fileName);
+					}
+
+					_messageBoxService.ShowInformationMessage(string.Format(PluginResources.ExecuteAnalyzeBatchTask_Message, fileName), PluginResources.InformativeLabel);
+				}
+
+				return false;
+			}
+			catch (Exception ex)
+			{
+				Log.Logger.Error($"ReportFileExist method: {ex.Message}\n {ex.StackTrace}");
+			}
+			return false;
+		}
+
+		/// <summary>
+		/// Check if the report folder exists
+		/// </summary>
+		/// <param name="projectInfoNode"></param>
+		/// <param name="projectXmlPath"></param>
+		/// <returns></returns>
+		public bool ReportFolderExist(XmlNode projectInfoNode, string projectXmlPath)
```

Comment:
This method should be refactored into two methods. One that sets the FoldersPath and one that actually checks if a filePath exists. This is because otherwise you create a coupling between the setting of the ReportsFolderPath and the check of the existence of a fileName, which is very counterintuitive. Someone who uses this method wouldn't know to call ReportFolderExist before trying to access the ReportsFolderPath property.

Reply:
Indeed, the setup of the file path will be extracted in a separate method, the changes will be reflected in the next commit